### ISSUE TITLE
Documentation in module twisted.words.protocols

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#6482]
|<img alt="Upasana's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| Upasana reported|
|-|-|
|Trac ID|trac#6482|
|Type|enhancement|
|Created|2013-04-26 22:06:25Z|

```
W9001:  1,0: Missing copyright header
W9002:  1,0: Missing a reference to test module in header
W9201:  1,0: The opening/closing of docstring should be on a line by themselves
```

Attachments:

* [bug_6482.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/3f3/3f3a57ecb273b0dc28f2204ff0b8f56f360563fa/71be33a4970d79de4374ea6fa9a978ba073f9f5f.patch) (407 bytes) - added by Upasana on 2013-04-26 22:09:22Z - 
* [bug_6482_2.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/3f3/3f3a57ecb273b0dc28f2204ff0b8f56f360563fa/5c340bc5df855e42a8936309d0cb5d505ad8bcb4.patch) (335 bytes) - added by Upasana on 2013-04-27 17:28:17Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__6482 6482
type__enhancement enhancement
reporter__Upasana Upasana
priority__low low
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__words words
keywords__easy easy
time__1367013985000000 1367013985000000
changetime__1367088661000000 1367088661000000
version__None None
owner__thijs thijs
cc__ralphm
```
</details>

**Comment by twisted-trac** [comment]
|<img alt="Upasana's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Upasana removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_3"></a>|@thijstriemstra set owner to Upasana|
|-|-|

Thanks for your contribution Upasana. We discussed the 2nd point of this ticket (W9002) on IRC and decided it's a bug in twistedcheckers.

```
<thijstriemstra> twistedchecker complains about a missing test header in a __init__.py file that only has a docstring. is that a bug or do we really want to have each module to have a header (refs #6482)
<tomprince> I'd suggest that is a twistedchecker bug.
<thijstriemstra> ok, ill open a ticket there
<tomprince> twistedchecker has a bunch of rules, but doesn't have reasonable exceptions to them.
```

Could you see if this issue is already reported on the https://launchpad.net/twistedchecker project and if it isn't, open a ticket there describing the issue?

And could you update the patch as well?

Thanks.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_5"></a>|@thijstriemstra set owner to @thijstriemstra<br>@thijstriemstra set status to `assigned`|
|-|-|

Thanks for your patch, I'll apply it. Did you open a ticket for twistedchecker btw?

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_6"></a>|@thijstriemstra set status to `closed`|
|-|-|

(In [38288]) Apply bug_6482_2.patch: Add missing copyright header and put the module docstring on 3 lines.

Author: Upasana
Reviewer: thijs
Fixes: [#6482](https://github.com/twisted/twisted/issues/6482)

**Comment by twisted-trac** [comment]
|<img alt="Upasana's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_7"></a>|Upasana commented|
|-|-|

Yes, I've done that, this is the URL of that bug: https://bugs.launchpad.net/twistedchecker/+bug/1173713
Thanks.
